sympy/functions/elementary/hyperbolic.py

```diff
@@ -1378,17 +1390,27 @@ def taylor_term(n, x, *previous_terms):
                 F = factorial(k)
                 return -R / F * S.ImaginaryUnit * x**n / n
 
-    def _eval_as_leading_term(self, x, logx=None, cdir=0):
-        from sympy.series.order import Order
-        arg = self.args[0].as_leading_term(x)
-
-        if x in arg.free_symbols and Order(1, x).contains(arg):
-            return S.ImaginaryUnit*S.Pi/2
-        else:
-            return self.func(arg)
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):  # acosh
+        arg = self.args[0]
+        x0 = arg.subs(x, 0).cancel()
+        # Handling Branch cuts (-oo, 1)
+        if x0 in (-S.One, S.Zero, S.One, S.ComplexInfinity):
+            return self.rewrite(log)._eval_as_leading_term(x, logx=logx, cdir=cdir)
+        ndir = arg.dir(x, cdir if cdir else 1)
+        if (x0 - 1).is_negative:
+            if im(ndir).is_negative:
+                if x0.is_negative:
+                    return self.func(x0) - 2*I*pi
+            elif im(ndir).is_positive:
+                if x0.is_positive:
+                    return self.func(x0)
```

Comment:
It seems that this block is redundant. It `x0` is not positive then `self.func(x0)` will be returned on the last line.

Reply:
Yeah correct , that's the case with some like `acos, asec, acosh` a couple others where leading term through one direction of approach at the branch cut would be `self.func(x0)` and from the other direction is `self.func(x0) - 2*I*pi` . So yeah I just thought of sticking to what might be technically correct (that would be to differentiate between both cases and return for both) but surely we can optimize here . I'll look into these cases and maybe just remove such blocks !